weblate/utils/tasks.py

```diff
@@ -83,27 +83,46 @@ def database_backup():
         return
     ensure_backup_dir()
     database = settings.DATABASES["default"]
-    if database["ENGINE"] != "django.db.backends.postgresql":
-        return
-    cmd = ["pg_dump", "--dbname", database["NAME"]]
-    if database["HOST"]:
-        cmd += ["--host", database["HOST"]]
-    if database["PORT"]:
-        cmd += ["--port", database["PORT"]]
-    if database["USER"]:
-        cmd += ["--username", database["USER"]]
-    if settings.DATABASE_BACKUP == "compressed":
-        cmd += ["--file", data_dir("backups", "database.sql.gz")]
-        cmd += ["--compress", "6"]
+    env = get_clean_env()
+
+    if database["ENGINE"] == "django.db.backends.postgresql":
+        cmd = ["pg_dump", "--dbname", database["NAME"]]
+
+        if database["HOST"]:
+            cmd += ["--host", database["HOST"]]
+        if database["PORT"]:
+            cmd += ["--port", database["PORT"]]
+        if database["USER"]:
+            cmd += ["--username", database["USER"]]
+        if settings.DATABASE_BACKUP == "compressed":
+            cmd += ["--file", data_dir("backups", "database.sql.gz")]
+            cmd += ["--compress", "6"]
+        else:
+            cmd += ["--file", data_dir("backups", "database.sql")]
+
+        env["PGPASSWORD"] = database["PASSWORD"]
+    elif database["ENGINE"] == "django.db.backends.mysql":
+        cmd = ["mysqldump", "--databases", database["NAME"]]
+
+        if database["HOST"]:
+            cmd += ["--host", database["HOST"]]
+        if database["PORT"]:
+            cmd += ["--port", database["PORT"]]
+        if database["USER"]:
+            cmd += ["--user", database["USER"]]
+        if settings.DATABASE_BACKUP == "compressed":
+            cmd += ["|", "gzip", ">", data_dir("backups", "database.sql.gz")]
```

Comment:
This doesn't work - the process is not executed in shell and these parameters are passed directly to `mysqldump`. Also removing this seems to unfreeze the test (though I'm not sure how is it related).

Reply:
Okay, so, compression off the table without a wrapper then, or just document that compression unsupported, or add shell=True?

Reply:
I've just added the shell option.  I don't think it's a big security issue in that all the parameters are under the admins control, and since the password, which is usually where you get into trouble, is passed through the environment, it should be pretty safe.

Either way, I defer to you as to how/if you want to support compression on MariaDB/MySQL.